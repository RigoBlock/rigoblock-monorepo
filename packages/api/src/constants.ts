export enum NETWORKS {
  MAINNET = '1',
  ROPSTEN = '3',
  KOVAN = '42',
  LOCALHOST = '5777'
}

export const RPC_URLS = {
  [NETWORKS.MAINNET]: 'wss://mainnet.infura.io/ws/',
  [NETWORKS.ROPSTEN]: 'wss://ropsten.infura.io/ws/',
  [NETWORKS.KOVAN]: 'wss://kovan.infura.io/ws/',
  [NETWORKS.LOCALHOST]: 'ws://localhost:8545'
}
