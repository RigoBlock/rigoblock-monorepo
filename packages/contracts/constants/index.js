const path = require('path')

const NETWORKS = {
  ganache: 'http://localhost:8545',
  mainnet: 'https://mainnet.infura.io/v3/d48872aa1c00471c825e9d856c3c3138',
  ropsten: 'https://ropsten.infura.io/v3/d48872aa1c00471c825e9d856c3c3138',
  kovan: 'https://kovan.infura.io/v3/d48872aa1c00471c825e9d856c3c3138'
}

module.exports = {
  TMP_DIR: '.tmp',
  ARTIFACTS_DIR: path.resolve(__dirname, '../artifacts'),
  CONTRACTS_DIR: path.resolve(__dirname, '../src'),
  NETWORKS,
  GANACHE_PORT: 8545,
  GANACHE_NETWORK_ID: 5777,
  GAS_ESTIMATE: 5e6,
  CONTRACT_NAMES: [
    'AEthfinex',
    'ATotlePrimary',
    'AWeth',
    'Authority',
    'Distribution',
    'Drago',
    'DragoEventful',
    'DragoFactory',
    'DragoRegistry',
    'ERC20',
    'Exchange',
    'ExchangeEfx',
    'ExchangeV1Fork',
    'ExchangesAuthority',
    'Faucet',
    'HGetDragoData',
    'Inflation',
    'Migrations',
    'NavVerifier',
    'Owned',
    'OwnedUninitialized',
    'ProofOfPerformance',
    'RigoToken',
    'SigVerifier',
    'TokenTransferProxy',
    'TotlePrimary',
    'UnlimitedAllowanceToken',
    'Vault',
    'VaultEventful',
    'VaultFactory',
    'WETH9',
    'WrapperLock',
    'WrapperLockEth',
    'ZeroExExchangeHandler'
  ]
}
