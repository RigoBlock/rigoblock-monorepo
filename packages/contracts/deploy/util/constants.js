export const constants = {
    optimizerEnabled: false,
    timeoutMs: 30000,
    contracts: '*' as '*'
}
