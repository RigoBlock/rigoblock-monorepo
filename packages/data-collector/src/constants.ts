export const REDIS_HOST = process.env.REDIS_HOST || '127.0.0.1'
export const REDIS_PORT = parseInt(process.env.REDIS_PORT) || 6379
export const REDIS_PASSWORD = process.env.REDIS_PASSWORD || ''
export const TELEGRAF_HOST = process.env.TELEGRAF_HOST || '127.0.0.1'
export const TELEGRAF_PORT = process.env.TELEGRAF_PORT || 8125

export const TOKEN_MARKET_BASE_URL =
  'https://tokenmarket.net/blockchain/ethereum/assets'
export const CRYPTO_NEWS_BASE_URL =
  'https://cryptopanic.com/api/posts/?auth_token=5b3f861c261e7e6243a55dc7b7ca600832156886&public=true&filter=hot&currencies='
