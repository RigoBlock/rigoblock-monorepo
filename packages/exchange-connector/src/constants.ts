export const MAINNET: string = 'mainnet'
export const ROPSTEN: string = 'ropsten'
export const KOVAN: string = 'kovan'
export const RINKEBY: string = 'rinkeby'

export const PRICE_PRECISION: number = 7
export const AMOUNT_PRECISION: number = 5
export const TO_WEI: number = 1e18

export enum supportedExchanges {
  ETHFINEX = 'Ethfinex',
  ETHFINEX_RAW = 'EthfinexRaw',
  ERCDEX = 'ERCdEX',
  ZEROEXRELAYER = '0xStandardRelayer'
}

export enum NETWORKS {
  MAINNET = 1,
  KOVAN = 42,
  ROPSTEN = 3,
  RINKEBY = 4
}

export enum WS_STATUS {
  CLOSED = 'closed',
  OPEN = 'open',
  CONNECTING = 'connecting',
  CLOSING = 'closing'
}
