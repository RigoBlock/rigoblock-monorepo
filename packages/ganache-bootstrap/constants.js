module.exports = {
  GANACHE_URL: 'http://localhost:8545'
}
