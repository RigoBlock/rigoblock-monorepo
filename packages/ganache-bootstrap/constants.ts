const task = {
  GANACHE_URL: 'http://localhost:8545'
}

export default task
