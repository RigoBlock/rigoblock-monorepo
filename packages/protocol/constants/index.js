const path = require('path')

const NETWORKS = process.env.NETWORKS
  ? process.env.NETWORKS.split(',')
  : ['http://localhost:8545']

module.exports = {
  ARTIFACTS_DIR: path.resolve(__dirname, '../artifacts'),
  CONTRACTS_DIR: path.resolve(__dirname, '../solidity-contracts'),
  NETWORKS,
  GANACHE_PORT: 8545,
  GANACHE_NETWORK_ID: 5777,
  GAS_ESTIMATE: 5e6,
  CONTRACT_NAMES: [
    'Authority',
    'Distribution',
    'Drago',
    'DragoEventful',
    'DragoFactory',
    'DragoRegistry',
    'ERC20',
    'Inflation',
    'Migrations',
    'Owned',
    'OwnedUninitialized',
    'ProofOfPerformance',
    'RigoToken',
    'TokenTransferProxy',
    'UnlimitedAllowanceToken',
    'Vault',
    'VaultEventful',
    'VaultFactory',
    'WrapperLock',
    'WrapperLockEth'
  ]
}
