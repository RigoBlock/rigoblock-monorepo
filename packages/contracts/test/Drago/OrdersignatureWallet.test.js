//import { BigNumber } from 'bignumber.js'
//import { ZeroEx } from '0x.js'
import { GANACHE_NETWORK_ID, GAS_ESTIMATE } from '../../constants'
import dragoArtifact from '../../artifacts/Drago.json'
import {
    assetDataUtils,
    BigNumber,
    ContractWrappers,
    generatePseudoRandomSalt,
    Order,
    orderHashUtils,
    signatureUtils,
    SignerType,
} from '0x.js'
import web3 from '../web3'

const contractName = 'Drago'

describeContract(contractName, () => {
  let dragoAddress
  let dragoInstance
  let transactionDefault
  let exchangeAddress
  //let hotWalletAddress
  let ethfinexAdapterAddress

  beforeAll(async () => {
    await baseContracts['DragoFactory'].createDrago('my new drago', 'DRA')
    const dragoData = await baseContracts['DragoRegistry'].fromName(
      'my new drago'
    )
    const [, address] = dragoData
    dragoAddress = address
    dragoInstance = new web3.eth.Contract(
      dragoArtifact.networks[GANACHE_NETWORK_ID].abi,
      dragoAddress
    )
    transactionDefault = {
      from: accounts[0],
      gas: GAS_ESTIMATE,
      gasPrice: 1
    }
    exchangeAddress = await baseContracts['Exchange'].address
    //hotWalletAddress = accounts[0] // hot wallet
    ethfinexAdapterAddress = await baseContracts[
      'ExchangesAuthority'
    ].getExchangeAdapter(exchangeAddress)
    await baseContracts['ExchangesAuthority'].setWhitelister(accounts[0], true)
  })

  describe('operateOnExchange', () => {
    it('logs an order signature for input data', async () => {
      const makerAddress = dragoAddress // '0x2f3ae8c5e7321688999883fd4f569e928d81d68f' // a drago
      const takerAddress = '0x0000000000000000000000000000000000000000'
      const senderAddress = '0xc2b5122381bcddb87e75fab2e46a70e7c19b69d3' // base account with GRG and ETH balance
      const feeRecipientAddress = '0x0000000000000000000000000000000000000000' // ZeroEx.NULL_ADDRESS
      const makerAssetData = '0xacfb4c79259e3c2c1bf054f136e6d75f7cc2b07e' // GRGW // TODO: double check format
      const takerAssetData = '0x06da2eb72279c1cec53c251bbff4a06fbfb93a5b' // ETHW // TODO: double check format
      //const exchangeAddress = exchangeAddress //'0x1d8643aae25841322ecde826862a9fa922770981'
      const salt = generatePseudoRandomSalt().toString()
      const makerFee = new BigNumber(0).toString()
      const takerFee = new BigNumber(0).toString()
      const makerAssetAmount = web3.utils.toWei('0.2').toString()
      const takerAssetAmount = web3.utils.toWei('0.3').toString()
      const expirationTimeSeconds = new BigNumber(
        Date.now() + 3600000
      ).toString() // Valid for up to an hour

      // Generate order
      const order = {
        exchangeAddress,
	      expirationTimeSeconds,
	      feeRecipientAddress,
	      makerAddress,
	      makerAssetAmount,
	      makerAssetData,
	      makerFee,
	      salt,
	      senderAddress,
	      takerAddress,
	      takerAssetAmount,
	      takerAssetData,
	      takerFee,
      }

      const orderHash = await orderHashUtils.getOrderHashHex(order)
      const provider = web3.currentProvider

      // Instantiate 0x.js instance
      const configs = {
        networkId: 50 // (1-mainnet, 3-ropsten, 4-rinkeby, 42-kovan, 50-testrpc)
      }
      //const zeroEx = new ZeroEx(provider, configs)

      // Signing orderHash -> ecSignature
      const signerAddress = accounts[0] //'0xc2b5122381bcddb87e75fab2e46a70e7c19b69d3' must include pkey

      const ecSignature = await signatureUtils.ecSignHashAsync(
        provider,
        orderHash,
        signerAddress
      )

      const v = new BigNumber(ecSignature.v).toString()
      const r = ecSignature.r
      const s = ecSignature.s

      const signature = ecSignature

      const typedSignature = await signatureUtils.convertToSignatureWithType(
        signature,
        'Wallet'
      )

      const data = orderHash

      // check against 0x api
      const isValidSignature = await expect(
        signatureUtils
        .isValidWalletSignatureAsync(
          provider,
          data,
          signature,
          dragoAddress
        )
      ).rejects.toThrowErrorMatchingSnapshot() // revert ORIGIN_NOT_WHITELISTED (caller must be hot wallet)

// TODO: check again by whitelisting the caller

/*
      // check against exchange smart contract
      const isValidSignature = await baseContracts['Exchange']
        .isValidSignature(
          data,
          dragoAddress, // the account that should have signed the transaction
          typedSignature,
        )
      expect(isValidSignature).toEqual(false)
      // VM Exception while processing transaction: revert SIGNATURE_UNSUPPORTED
*/
    })
  })
})
