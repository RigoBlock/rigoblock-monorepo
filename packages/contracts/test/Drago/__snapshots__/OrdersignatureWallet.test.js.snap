// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Drago operateOnExchange logs an order signature for input data 1`] = `"VM Exception while processing transaction: revert ORIGIN_NOT_WHITELISTED"`;
